src/Compilers/CSharp/Portable/Symbols/AliasSymbol.cs

```diff
@@ -249,7 +249,7 @@ internal NamespaceOrTypeSymbol GetAliasTarget(ConsList<Symbol> basesBeingResolve
 
                 if ((object)Interlocked.CompareExchange(ref _aliasTarget, symbol, null) == null)
                 {
-                    bool won = ImmutableInterlocked.InterlockedInitialize(ref _aliasTargetDiagnostics, newDiagnostics.ToReadOnlyAndFree());
+                    bool won = Interlocked.Exchange(ref _aliasTargetDiagnostics, newDiagnostics) == null;
                     Debug.Assert(won, "Only one thread can win the alias target CompareExchange");
 
```

Comment:
This Assert seems to check that CompareExchange works as expected. 
Perhaps the whole thing should be just " _aliasTargetDiagnostics = newDiagnostics "